### PR TITLE
constants: add RTSP/RAOP methods

### PR DESCRIPTION
**Pull request by indutny**
Supersedes: #66 

cc @nodejs/http

### CODE REVIEW

**Comment by indutny** [comment]
Benchmarks:

**with** change:
```
http loose: "seanmonstar/httparse" (C)
8192.00 mb | 1529.18 mb/s | 2280879.50 ops/sec | 5.36 s
http strict: "seanmonstar/httparse" (C)
8192.00 mb | 1414.58 mb/s | 2109943.20 ops/sec | 5.79 s
http loose: "nodejs/http-parser" (C)
8192.00 mb | 1044.18 mb/s | 2105571.59 ops/sec | 7.85 s
http strict: "nodejs/http-parser" (C)
8192.00 mb | 1004.79 mb/s | 2026150.04 ops/sec | 8.15 s
```

**without** change:
```
http loose: "seanmonstar/httparse" (C)
8192.00 mb | 1559.15 mb/s | 2325589.03 ops/sec | 5.25 s
http strict: "seanmonstar/httparse" (C)
8192.00 mb | 1402.73 mb/s | 2092269.69 ops/sec | 5.84 s
http loose: "nodejs/http-parser" (C)
8192.00 mb | 1081.31 mb/s | 2180449.94 ops/sec | 7.58 s
http strict: "nodejs/http-parser" (C)
8192.00 mb | 1027.44 mb/s | 2071824.99 ops/sec | 7.97 s
```

**Comment by pallas** [comment]
Thanks, looking.

**Comment by pallas** [comment]
After this change, I'm getting HPE_INVALID_CONSTANT again.. going to investigate why.

File: src/llhttp/constants.ts [hunk 1]
```diff
@@ -113,6 +113,64 @@ export enum METHODS {
...
+
+export const METHODS_RTSP = [
+  METHODS.OPTIONS,
+  METHODS.DESCRIBE,
+  METHODS.ANNOUNCE,
+  METHODS.SETUP,
+  METHODS.PLAY,
+  METHODS.PAUSE,
+  METHODS.TEARDOWN,
+  METHODS.GET_PARAMETER,
```
Comment (pallas) [inline src/llhttp/constants.ts:168]: SET_PARAMETER is missing

Comment (indutny) [reply src/llhttp/constants.ts:168]: Ooops. Should be fixed now!

**Comment by pallas** [comment]
Ah, I see my current problem. It looks like Apple sends GETs in addition to the listed verbs.

**Comment by pallas** [comment]
Ok, can you please add GET and POST to the RTSP verbs? After doing so, this works great with AirPlay.

**Comment by indutny** [comment]
@pallas sure thing, done!

**Comment by pallas** [comment]
Verified that this works perfectly for AirPlay.

**Comment by indutny** [comment]
Landed in 0b0772cda172098651c332e008c15180880a4f47 and 07edcd273d8e6423f3e04f18f35214bcb3a48351. Thanks for checking it!

**Comment by indutny** [comment]
...and released as [2.2.0](https://github.com/nodejs/llhttp/releases/tag/v2.2.0)

**Pull request closed without merging**